runtime/autoload/man.vim

```diff
@@ -57,30 +53,77 @@ function! man#open_page(count, count1, mods, ...) abort
     noautocmd execute 'silent' a:mods 'split' fnameescape(bufname)
   endif
   let b:man_sect = sect
-  call s:read_page(path)
+  call s:put_page(page)
 endfunction
 
 function! man#read_page(ref) abort
   try
     let [sect, name] = man#extract_sect_and_name_ref(a:ref)
     let [b:man_sect, name, path] = s:verify_exists(sect, name)
+    let page = s:get_page(path)
   catch
     " call to s:error() is unnecessary
     return
   endtry
-  call s:read_page(path)
+  call s:put_page(page)
+endfunction
+
+" Handler for s:system() function.
+function! s:system_handler(jobid, data, event) dict abort
+  if a:event == 'stdout'
+    let self.stdout .= join(a:data, "\n")
+  elseif a:event == 'stderr'
+    let self.stderr .= join(a:data, "\n")
+  else
+    let self.exit_code = a:data
+  endif
+endfunction
+
+" Run a system command and timeout after 30 seconds.
+function! s:system(cmd, ...) abort
+  let opts = {
+        \ 'stdout': '',
+        \ 'stderr': '',
+        \ 'exit_code': 0,
+        \ 'on_stdout': function('s:system_handler'),
+        \ 'on_stderr': function('s:system_handler'),
```

Comment:
@justinmk, the reason s:system() is defined the way it is in man.vim is so that health/provider.vim could also use it. I worked on a patch yesterday to do this, but i'm not sure if I should create another PR just for this patch since it is a big refactor. (I had to change the way error reporting was gathered. [return values vs exceptions) The only reason to include the patch in this PR is to fix the fact that I removed a 2>/dev/null from line 419 in health/provider.vim.